favorites_db.py

```diff
@@ -0,0 +1,72 @@
+from peewee import *
+
+db = SqliteDatabase('favorites.db')
+
+class Favorite(Model):
+
+    city = CharField(null=False)
+    country = CharField(null=False)
+    month = IntegerField(null=False, constraints=[Check('0 < month AND month < 13')])
+    year = IntegerField(null=False)
+    webcam = CharField(null=True) # for now I allow null values - in case we got nothing from one of the APIs, but user still wants to save favorite with remaining info
+    weather = CharField(null=True) # however we could change this to a string that says 'no data found' instead if wanted?
+    holidays = CharField(null=True)
+    nickname = CharField(null=True) # in case we want to allow the user to give their favorite a nickname
+
+    class Meta: 
+        database = db
+
+    def __str__(self):
+        return f'ID {self.id}: {self.city}, {self.country}, in {self.month}/{self.year}: Webcam: {self.webcam}, Weather: {self.weather}, Holidays: {self.holidays}, Nickname: {self.nickname}'
+
+
+class FavoritesError(Exception):
+    """Custom exception class.""" 
+    pass
+
+
+def create_table():
+    """Create Favorites table."""
+    db.create_tables([Favorite])
+
+
+def get_favorites():
+    """Select all favorites from the favorites table.
+    Return a list of favorites, or an empty list."""
+    # TODO: handle an empty list elsewhere in the program
+    # since we don't want there to be an error if the user has simply not added anything to faves yet.
+    # want html to instead display something like 'nothing in favorites list, try adding something to favorites'
+    favorites = Favorite.select()
```

Comment:
You can also use the list constructor and save a little code here
`favorites_list = list(favorites)`

Reply:
Done, thanks!